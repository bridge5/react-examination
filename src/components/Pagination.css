:local(.pagination) {
  padding: 10px 0;
  display: flex;
  justify-content: center;
  list-style-type: none;
}

:local(.page) {
  padding: 0 10px;
  cursor: pointer;
}

:local(.current) {
  color: white;
  cursor: default;
}
