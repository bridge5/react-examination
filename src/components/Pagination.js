import React, { Component } from 'react';
import PropTypes from 'prop-types';
import styles from './Pagination.css';
import PaginationItem from './PaginationItem';

class Pagination extends Component {
  componentDidMount() {
    const { currentPage } = this.props;
    this.gotoPage(currentPage);
  }

  render() {
    const { currentPage } = this.props;
    const totalPages = this.getTotalPages();

    return (
      <nav>
        <ul className={styles.pagination}>
          {totalPages.map((page, index) => 
            <PaginationItem
              key={index} 
              currentPage={currentPage}
              gotoPage={this.gotoPage}
              page={page} />
          )}
        </ul>
      </nav>
    );
  }

  gotoPage = page => {
    const { onPageChanged, perListItems } = this.props;
    const totalPages = this.getTotalPages();

    const paginationData = {
      currentPage: page,
      totalPages,
      perListItems
    };

    onPageChanged(paginationData)
  };

  getTotalPages = () => {
    const { totalItems, perListItems } = this.props
    const reminder = totalItems % perListItems;
    let pages = parseInt(totalItems / perListItems, 10);
    if (reminder) {
      pages = pages + 1;
    }
    const totalPages = Array.from({ length: pages }, (v, i) => i + 1);
    return totalPages;
  };
}

Pagination.propTypes = {
  totalItems: PropTypes.number.isRequired,
  currentPage: PropTypes.number,
  perListItems: PropTypes.number,
  onPageChanged: PropTypes.func.isRequired,
};

Pagination.defaultProps = {
  currentPage: 1,
  perListItems: 3
};

export default Pagination;
