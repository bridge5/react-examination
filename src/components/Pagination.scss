.pagination-wrapper {
  border-top: 1px solid #f2f2f5;
  margin-top: 20px;
  ul {
      height: 26px;
      padding: 4px 0;
      line-height: 26px;
      text-align: center;
      li {
          display: inline-block;
          a {
              display: inline-block;
              color: #333;
          }
          &.page-pre,
          &.page-after {
              width: 100px;
              a {
                  cursor: pointer;
                  padding: 0 10px;
                  &.off {
                      cursor: default;
                      color: #aaa;
                  }
                  &:hover {
                      background: #f2f2f5;
                      &.off {
                          background: none;
                      }
                  }
              }
          }
          &.page-count {
              box-sizing: border-box;
              border-left: 1px solid #f2f2f5;
              border-right: 1px solid #f2f2f5;
              border-left: 1px solid #d9d9d9;
              border-right: 1px solid #d9d9d9;
              line-height: 100%;
              a {
                  padding: 0 7px;
                  height: 20px;
                  line-height: 20px;
                  margin: 0 8px;
                  border-radius: 2px;
                  &.on {
                    //   background: #f2f2f5;
                    background-color: #FFF;
                      font-size: 20px;
                  }
                  &:hover {
                      background: #f2f2f5;
                      cursor: pointer;
                      &.disable {
                          background: none;
                          cursor: default;
                      }
                  }
              }
              &.alone {
                  border: none;
              }
          }
      }
  }
}