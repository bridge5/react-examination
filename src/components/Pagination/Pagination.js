import React from 'react';
import styles from './Pagination.css';
const { pageContainer } = styles;
export default ({ currentPageIndex, changePage, totalPageIndex }) => {
  return (
    <div className={pageContainer}>
      {currentPageIndex <= 0 ? null : (
        <button className="btn btn-default" onClick={() => changePage(-1)}>
          &laquo;
        </button>
      )}
      <button className="btn btn-default">{currentPageIndex + 1}</button>
      {currentPageIndex === totalPageIndex ? null : (
        <button className="btn btn-default" onClick={() => changePage(1)}>
          &raquo;
        </button>
      )}
    </div>
  );
};
