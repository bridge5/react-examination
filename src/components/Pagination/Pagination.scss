:local(.pageContainer) {
  display: flex;
  justify-content: center;
  align-items: center;
  padding: 10px 0;
  width: 100%;
}
