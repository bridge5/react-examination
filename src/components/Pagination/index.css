.pagination {
  display: block;
  box-sizing: border-box;
  margin: 0;
  padding: 0;
  color: rgba(0, 0, 0, 0.65);
  font-size: 14px;
  font-variant: tabular-nums;
  line-height: 1.5715;
  list-style: none;
  font-feature-settings: "tnum";
}

.pagination-item {
  display: inline-block!important;
  text-align: center;
  min-width: 24px;
  height: 24px;
  margin: 0;
  line-height: 22px;
  list-style: none;
  border: 1px solid #d9d9d9;
  border-radius: 2px;
  outline: 0;
  cursor: pointer;
}

.pagination-item:hover {
  color: #1890ff;
}

.pagination-item--active {
  font-weight: 500;
  background: #fff;
  border-color: #1890ff;
  color: #1890ff;
}
