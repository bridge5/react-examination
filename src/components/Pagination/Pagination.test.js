import React from 'react';
import { shallow } from 'enzyme';
import Pagination from './Pagination';

describe('Pagination feature', () => {
  it('render Paginator correctly', () => {
    const mockChangePage = jest.fn();
    const wrapper = shallow(
      <Pagination
        currentPageIndex={1}
        totalPageIndex={2}
        changePage={mockChangePage}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
