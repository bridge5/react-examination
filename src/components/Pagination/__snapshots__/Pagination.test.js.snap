// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination feature render Paginator correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <_default
    changePage={[MockFunction]}
    currentPageIndex={1}
    totalPageIndex={2}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <button
          className="btn btn-default"
          onClick={[Function]}
        >
          «
        </button>,
        <button
          className="btn btn-default"
        >
          2
        </button>,
        <button
          className="btn btn-default"
          onClick={[Function]}
        >
          »
        </button>,
      ],
      "className": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "«",
          "className": "btn btn-default",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "«",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": 2,
          "className": "btn btn-default",
        },
        "ref": null,
        "rendered": 2,
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "»",
          "className": "btn btn-default",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "»",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <button
            className="btn btn-default"
            onClick={[Function]}
          >
            «
          </button>,
          <button
            className="btn btn-default"
          >
            2
          </button>,
          <button
            className="btn btn-default"
            onClick={[Function]}
          >
            »
          </button>,
        ],
        "className": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "«",
            "className": "btn btn-default",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "«",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": 2,
            "className": "btn btn-default",
          },
          "ref": null,
          "rendered": 2,
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "»",
            "className": "btn btn-default",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "»",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
